tests/components/websocket_api/test_connection.py

```diff
@@ -20,3 +26,32 @@ async def send_big_result(hass, connection, msg):
     assert msg["type"] == const.TYPE_RESULT
     assert msg["success"]
     assert msg["result"] == {"big": "result"}
+
+
+async def test_exception_handling():
+    """Test handling of exceptions."""
+    send_messages = []
+    conn = websocket_api.ActiveConnection(
+        logging.getLogger(__name__), None, send_messages.append, None, None
+    )
+
+    for (exc, code, err) in (
+        (exceptions.Unauthorized(), websocket_api.ERR_UNAUTHORIZED, "Unauthorized"),
+        (
+            vol.Invalid("Invalid something"),
+            websocket_api.ERR_INVALID_FORMAT,
+            "Invalid something. Got {'id': 5}",
+        ),
+        (asyncio.TimeoutError(), websocket_api.ERR_TIMEOUT, "Timeout"),
+        (
+            exceptions.HomeAssistantError("Failed to do X"),
+            websocket_api.ERR_UNKNOWN_ERROR,
+            "Failed to do X",
+        ),
+        (ValueError("Really bad"), websocket_api.ERR_UNKNOWN_ERROR, "Unknown error"),
+    ):
+        send_messages.clear()
+        conn.async_handle_exception({"id": 5}, exc)
+        assert len(send_messages) == 1
+        assert send_messages[0]["error"]["code"] == code
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)